examples/colour-uniform/main.rs

```diff
@@ -140,14 +140,23 @@ struct RendererState<B: Backend> {
     swapchain: Option<SwapchainState<B>>,
     device: Rc<RefCell<DeviceState<B>>>,
     backend: BackendState<B>,
-    window: WindowState,
     vertex_buffer: BufferState<B>,
     render_pass: RenderPassState<B>,
     uniform: Uniform<B>,
     pipeline: PipelineState<B>,
     framebuffer: FramebufferState<B>,
     viewport: pso::Viewport,
     image: ImageState<B>,
+    recreate_swapchain: bool,
+    r: f32,
```

Comment:
can we use [ColorValue](https://docs.rs/gfx-hal/0.2.1/gfx_hal/pso/type.ColorValue.html) for these?